### PR TITLE
fix: [ANDROAPP-6013] don't save events without date or catCombo

### PR DESCRIPTION
**Pull request by mmmateos**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/bindings/ValidationStrategyExtensions.kt [hunk 1]
```diff
@@ -1,9 +1,18 @@
 package org.dhis2.bindings
 
+import org.dhis2.form.model.EventMode
 import org.hisp.dhis.android.core.common.ValidationStrategy
 
-fun ValidationStrategy.canSkipErrorFix(hasErrorFields: Boolean, hasEmptyMandatoryFields: Boolean) =
+fun ValidationStrategy.canSkipErrorFix(
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/bindings/ValidationStrategyExtensions.kt:6]: I think that this extensions it's only used in `EventCapturePresenterImpl` why don't we refactor it as a private fun into the presenter?

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.kt [hunk 2]
```diff
@@ -109,6 +111,7 @@ class EventCapturePresenterImpl(
         errorFields: List<FieldWithIssue>,
         emptyMandatoryFields: Map<String, String>,
         warningFields: List<FieldWithIssue>,
+        eventMode: EventMode?,
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.kt:114]: is this `attemptFinish` method covered by test all the possible outcomes?, maybe we should do it in order to validate all the expected outputs

File: app/src/test/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventIntegrationTest.kt [hunk 3]
```diff
@@ -119,13 +120,9 @@ class EventIntegrationTest {
             onCompleteMessage = null,
             allowDiscard = false,
         )
-        eventCaptureFormPresenter.handleDataIntegrityResult(dataCheckResult)
+        eventCaptureFormPresenter.handleDataIntegrityResult(dataCheckResult, EventMode.CHECK)
```
Comment (andresmr) [inline app/src/test/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventIntegrationTest.kt:123]: Could we add test for EventMode.NEW?

File: form/src/main/java/org/dhis2/form/data/FormValueStore.kt [hunk 4]
```diff
@@ -63,14 +63,18 @@ class FormValueStore(
...
                         EVENT_REPORT_DATE_UID,
                         ValueStoreResult.VALUE_CHANGED,
-                    ),
+                    )
+                } ?: StoreResult(
+                    EVENT_REPORT_DATE_UID,
+                    ValueStoreResult.VALUE_HAS_NOT_CHANGED,
                 )
+
+                Flowable.just(storeResult)
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/data/FormValueStore.kt:77]: Should we do the same in the org unit field instead of the try{} catch {}?

**Review by xavimolloy — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**